core-primitives/enclave-api/src/enclave_base.rs

```diff
@@ -125,22 +123,13 @@ impl EnclaveBase for Enclave {
 
 	fn init_light_client<SpHeader: Header>(
 		&self,
-		genesis_header: SpHeader,
-		authority_list: VersionedAuthorityList,
-		authority_proof: Vec<Vec<u8>>,
+		params: LightClientInitParams<SpHeader>,
 	) -> EnclaveResult<SpHeader> {
-		let encoded_genesis_header = genesis_header.encode();
-		let authority_proof_encoded = authority_proof.encode();
-
+		let encoded_params = params.encode();
 		// Todo: this is a bit ugly but the common `encode()` is not implemented for authority list
-		let latest_header_encoded = authority_list.using_encoded(|authorities| {
-			init_light_client_ffi(
-				self.eid,
-				authorities.to_vec(),
-				encoded_genesis_header,
-				authority_proof_encoded,
-			)
-		})?;
+		let latest_header_encoded = params
+			.get_authorities()
+			.using_encoded(|_authorities| init_light_client_ffi(self.eid, encoded_params))?;
```

Comment:
This looks very fancy, but why is this piece of code here instead of:

```
let latest_header_encoded = init_light_client_ffi(self.eid, params.encode())?;
```